blog/2023-07-3.4-3.13.md

```diff
@@ -0,0 +1,191 @@
+
+<img src="/img/blog/3.4-3.13/10k-files.jpg" className="border" alt="Uppy UI with over 10k files selected from a remote source" />
+
+We’ve also fixed an issue with delayed throttled progress events, which could lead to incorrect progress.
+
+## Transloadit
+
+### assemblyOptions
+
+We’ve introduced a new option called [`assemblyOptions: object | function`](https://uppy.io/docs/transloadit/#assemblyoptions), which replaces the `getAssemblyOptions`, `params` and `fields` (those are now deprecated and will be removed in the next major).
+
+Here’s how you can use the new option as an object:
+
+```js
+uppy.use(Transloadit, {
+  assemblyOptions: {
+    params: {
+      auth: { key: 'key-from-transloadit' },
+      template_id: 'id-from-transloadit',
+      steps: {
+        // Overruling Template at runtime
+      },
+    },
+    signature: 'generated-signature',
+    fields: {
+      // Dynamic or static fields to send along
+    },
+  }
+})
+```
+
+And here’s an example with a function, to be able to set meta fields per-file, for instance:
+
+```js
+uppy.use(Transloadit, {
+  assemblyOptions (file) {
+    return {
+      params: {
+        auth: { key: 'TRANSLOADIT_AUTH_KEY_HERE' },
+        template_id: 'xyz',
+      },
+      fields: {
+        caption: file.meta.caption,
+      },
+    }
+  }
+})
+```
+
+### Server-sent events
+
+Historically Transloadit supported progress updates via Socket.io. It is a robust and stable package, but came with a “price-tag” in the form of bundle size: [249.3 kB Minified](https://bundlephobia.com/package/socket.io@4.7.1). So it’s been on our minds for a while to replace it with something more lightweight, without breaking backwards-compatibility for the older clients (so simply removing Socket.io on the server in favor of plain WebSockets is not desireable).
+
+The answer — [Server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events). It’s a simple low-overhead one-way connection from Transloadit servers to `@uppy/transloadit`. As was the case with Socket.io, we fall back to HTTP polling every 2 seconds if SSE connection fails for some reason.
+
+Socket.io is still around in `@uppy/transloadit`, but will be removed soon, once SSE proves itself in production.
+
+### Bug fixes
+
+- Reset `tus` key in the file on error, so retried files are re-uploaded.
+- Clean up event listener to prevent cancelled assemblies.
+- Ensure `fields` is not nullish when there are no files in Assembly.
+
+## Companion
+
+### Refresh tokens
+
+@uppy/companion: implement refresh for authentication tokens (#4448)
+
+### Bug fixes and improvements
+
+- Fixed 500 when file name contains non-ASCII chars
```

Comment:
Not sure what case this is referring to

Reply:
https://github.com/transloadit/uppy/pull/4493

Reply:
I'll rephrase.